core/src/main/java/org/geysermc/geyser/translator/protocol/bedrock/BedrockContainerCloseTranslator.java

```diff
@@ -70,6 +70,11 @@ public void translate(GeyserSession session, ContainerClosePacket packet) {
                         NetworkStackLatencyPacket latencyPacket = new NetworkStackLatencyPacket();
                         latencyPacket.setFromServer(true);
                         latencyPacket.setTimestamp(MAGIC_VIRTUAL_INVENTORY_HACK);
+                        session.getLatencyPingCache().add(() -> {
+                            if (session.getPendingOrCurrentBedrockInventoryId() != -1) {
+                                InventoryUtils.openPendingInventory(session);
+                            }
+                        });
                         session.sendUpstreamPacket(latencyPacket);
```

Comment:
Since we now always need to add to the ping cache when sending this packet, we should add a method to the session that sends this packet with provided parameters (and adds to the cache)